module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -211,6 +211,25 @@ public function setParameters($parameters = [])
         }
     }
 
+    /**
+     * Checks whether the search model is valid for the EDS API.
+     *
+     * @return bool Returns false if any model parameters are invalid.
+     */
+    public function isValid()
+    {
+        $contentProviderValues = $this->facetFilters['ContentProvider'] ?? [];
+        $invalidContentProviderCharacters = '|';
```

Comment:
Hi @cwolfebsco.  I'm making the assumption here that an EDS API content provider string will never include a `|` character.  When I do an empty search, looking through all of the providers listed, this is true -- there's nothing but alphanumeric, ampersands, whitespace, and a few with parens, colons, periods and hyphens i.e. `Applied Science & Technology Index Retrospective: 1913-1983 (H.W. Wilson)`.  But there are of course many databases that my institution doesn't license.  Is this still a safe assumption?

Reply:
Good Morning - I am currently on vacation and can't check on this properly but I am unaware of any database that would contain a pipe character. At the same time I don't have perfect visibility into that and will need to reach out to someone about it. I'll do that after my vacation is over (after Easter)

Reply:
@maccabeelevine  - After enjoying the great outdoors in Utah & Arizona (amazing National Parks, great hiking), I am back at work and as promised and searched our database lists for occurrences of the pipe character (|).

**Good news**, there are no subscription databases that contain a pipe character.

**Not great news**, there are very small number of custom catalog databases, which contain a pipe character. However since these are customer specific databases they are unlikely to show up in VuFind.

Reply:
Thanks Claus!  Sounds like an awesome vacation.

I'm taking this as good news.  If the customers with these unique databases are already unlikely to show up in VuFind (which makes sense), I think they are even less likely to be using the EDS API blended with ProQuest and thus turning on validation in the first place.  So I don't think this is worth making the configuration any more complex.  I thought of raising `$invalidContentProviderCharacters` to be a class attribute, but I decided even that is a bad idea as if we *do* want to load it from config in the future it might be an array, so better not to decide that now.  

For now I did add comments to EDS.ini and SearchRequestModel.

Reply:
I agree that this seems pretty safe. And as you say, if we run into a real use case where all of these rare things occur, we can likely address it by adding some more flexible configuration settings (e.g. a validation regex, or an exception list for specific database names). But as you say, no point in over-engineering this prematurely without a use case.